azure-kusto-data/README.md

```diff
@@ -1,19 +1,63 @@
 # Microsoft Azure Kusto Data Library for Node
 
-# Quick Start
+## Installation
+
+`npm install azure-kusto-data`
+
+## Quick Start
 
 ```javascript 
 const KustoClient = require("azure-kusto-data").Client;
 const KustoConnectionStringBuilder = require("azure-kusto-data").KustoConnectionStringBuilder;
-const Console = require("console");
 
-const kcsb = KustoConnectionStringBuilder.withAadApplicationKeyAuthentication(`https://${clusterName}.kusto.windows.net`,'username','password','tenant_id');
+const kcsb = KustoConnectionStringBuilder.withAadApplicationKeyAuthentication(`https://${clusterName}.kusto.windows.net`,'appid','appkey','authority_id');
 const client = new KustoClient(kcsb);
 
 client.execute("db", "TableName | limit 1", (err, results) => {
     if (err) throw new Error(err);
-    Console.log(JSON.stringify(results));
-    Console.log(results.primaryResults[0].toString());
+    console.log(JSON.stringify(results));
+    console.log(results.primaryResults[0].toString());
 });
 
-```
\ No newline at end of file
+```
+
+## Authentication
+There are several authentication methods
+
+### App
+The recommended way to authenticate is to use app id and key
+
+```javascript
+const kcsb = KustoConnectionStringBuilder.withAadApplicationKeyAuthentication(`https://${clusterName}.kusto.windows.net`,'appid','appkey','authorityId');
+```
+
+It is also possible to use a certificate:
+
+```javascript
+const kcsb = KustoConnectionStringBuilder.withAadApplicationCertificateAuthentication(`https://${clusterName}.kusto.windows.net`, 'appid', 'certificate', 'thumbprint', 'authorityId');
+```
+
+
+### Username/Password
+```javascript
+KustoConnectionStringBuilder.withAadUserPasswordAuthentication(`https://${clusterName}.kusto.windows.net`,'username','password');
+```
+
+Authority is optional, as it is inferd from the domain ('user@microsoft.com' would make the authority 'microsoft.com'). 
```

Comment:
It is not always true. Authority is optional when it can be inferred from the domain of the upn.